cmake/sycl/CMakeSYCLInformation.cmake

```diff
@@ -8,12 +8,13 @@
 include( CMakeCommonLanguageInclude )
 
 # Set up platform specific flags.
-include( Platform/${CMAKE_EFFECTIVE_SYSTEM_NAME}-IntelLLVM-SYCL OPTIONAL )
+include( Platform/${CMAKE_EFFECTIVE_SYSTEM_NAME}-${CMAKE_SYCL_COMPILER_ID}-SYCL
+   OPTIONAL )
 
 # Set up how SYCL object file compilation should go.
 if( NOT DEFINED CMAKE_SYCL_COMPILE_OBJECT )
    set( CMAKE_SYCL_COMPILE_OBJECT
-      "<CMAKE_SYCL_COMPILER> <DEFINES> <INCLUDES> <FLAGS> -o <OBJECT> -c <SOURCE>" )
+      "<CMAKE_SYCL_COMPILER> -x c++ <DEFINES> <INCLUDES> <FLAGS> -o <OBJECT> -c <SOURCE>" )
```

Comment:
I may be wrong, but this looks dangerous in a hipSYCL context, where from memory it may inadvertently turn off HIP/CUDA support in the underlying device compiler behind the frontend.

Reply:
Unfortunately there's not really a better solution... Without it, the compiler doesn't know what to do with a file called `.sycl`.

What's worse is that ComputeCpp for instance uses `.sycl` as the extension for its pre-compiled headers.

And no, we can't just use `.cxx` as the extension for the SYCL files, because then we can't have CMake treat these files as a "separate language".

So yeah, we do need to be careful with this. Luckily both ComputeCpp and hipSYCL seem to be smart enough to not propoagate this flag to their internal compilation commands. It just instructs them about the file given to them. Internally they use their own flags to process that file correctly. (For instance when you add this flag to `SYCLFLAGS`, that would add it to all of the compilation and linking commands. Which definitely does make things break. I tried...)

Reply:
I suggest going forward with the code as-is for now. We can tweak things later on, once we/I figure out how to make hipSYCL and ComputeCpp able to not only compile, but to also run the tests...